swap_meet/vendor.py

```diff
@@ -1,2 +1,93 @@
+from typing import ItemsView
+
+
 class Vendor:
-    pass
\ No newline at end of file
+    #Wave 1
+    ''' We are creating an inventory, which is an empty list, for storing,removing and adding items'''
+    def __init__(self,inventory=None):
+        if inventory is None:
+            inventory = []
+        self.inventory = inventory
+
+    def add(self,item):
+        self.inventory.append(item)
+        return item
+
+
+    def remove(self,item):
+        if item not in self.inventory:
+            return False
+        else:
+            self.inventory.remove(item)
+            return item
+    
+    #Wave 2
+    '''Returns a list of items in the inventory with that category'''
+    def get_by_category(self, category):
+        items = []
+        for item in self.inventory:
+            if item.category == category:
+                items.append(item)
+        return items
+    
+    #Wave 3
+    '''Swap item between self inventory and friend inventory, returns a boolean'''
+    def swap_items(self, friend_vendor, my_item, their_item):
+        if my_item in self.inventory and their_item in friend_vendor.inventory:
+            self.add(their_item)
+            friend_vendor.add(my_item)
+            self.remove(my_item)
+            friend_vendor.remove(their_item)
+            return True
+        return False
+    
+    #Wave 4
+    '''Swap the first item, returns a boolean.'''
+    def swap_first_item(self,friend_vendor):
+        if len(self.inventory) == 0 or len(friend_vendor.inventory) == 0:
+            return False
+        else:
+            return self.swap_items(friend_vendor,self.inventory[0],friend_vendor.inventory[0])
```

Comment:
Ahhh, I love this! Yes, make that `swap_items()` helper method do some of the boolean logic! 👍🏻